### DISCUSSION TITLE
Entry points (pyproject.toml) using self-written script?

### DISCUSSION THREAD

**Discussion by buhtz** [pypa/setuptools#3729] [Q&A] [answered]
Setuptools do create a script for each [entry point](https://setuptools.pypa.io/en/latest/userguide/entry_point.html) by default.
Is it possible to tell setuptools not to use that default script but my own one? Or is it possible to modify the script generated by setuptools?

My users can start my application usually as normal users but also as root. For the latter I use policy-kit (`pkexec`) which needs a special start script like this

```
#!/bin/sh

if [ "x$XDG_SESSION_TYPE" = "xwayland" ]; then
    PREFIX="env QT_QPA_PLATFORM=wayland-egl XDG_RUNTIME_DIR=$XDG_RUNTIME_DIR"
else
    PREFIX=""
fi

pkexec --disable-internal-agent $PREFIX "/usr/bin/backintime-qt" "$@"
```

There are also some other situations and cases where I maybe need to use my own script. Sometimes I have to modify/set environment variables before the application starts.

Can I achive this with entry_points in pyproject.toml?

**Comment by abravalheri** [comment]
Hi @buhtz , the executable wrappers for the entry points are created by `pip` with the help of `distlib`. `setuptools` is no longer used in this process.

I imagine that it is a default template and cannot be changed.

Maybe you can try creating a script that spwans a subprocess and then sets up the correct env vars?

**Reply by buhtz** [reply]
> executable wrappers for the entry points are created by `pip` with the help of `distlib`.

OK, I open a question about it at the [pip forum](https://discuss.python.org/t/pip-modify-or-replace-the-entry-point-sripts/22007)

**Comment by buhtz** [comment, marked as answer]
Please find example code demonstrating the solution [here](https://codeberg.org/buhtz/bit_demo).
